### ISSUE TITLE
2.4.11 checkered or dashed backgrounds can pass without being discernible

### ISSUE THREAD

**Issue by JasonAment** [w3c/wcag#1061]
[Codepen to illustrate this issue](https://codepen.io/JasonAment/pen/qBdjPeP)

The Codepen illustrates that a focus indicator can just shift the light and dark pixels (meeting the 3:1 contrast ratio) of a border to meet the requirement.  This is an imperceptible change.

Since each stripe or checkered piece occupies a different location that was previously the opposite color, this will meet all of the requirements.

**Minimum area:** The focus indication area is a 6px border on all edges.
**Focus contrast:** Color changes from each checker or stripe are a 3:1 contrast ratio with each other (#B80000/#000000 = 3:1)
**Contrast or thickness:** The focus indication area has a 3:1 contrast ratio against all adjacent colors for the minimum area or greater (both the black and red colors meet contrast ratios against the white), or has a thickness of at least 2 CSS pixels (thickness is a minimum of 6 pixels).

I wonder if the term "visible" could be changed to "discernible".

**Comment by JAWS-test** [comment]
Your example is a good proof of the current gaps in SC. But with what wording could these gaps to be closed?

> I wonder if the term "visible" could be changed to "discernible".

"discernible" is a subjective term that cannot be tested consistently.

**Comment by JasonAment** [comment]
@JAWS-test - I agree that discernible and visible are both subjective terms. Perhaps using the "distinguishing a visual element" language like 1.4.1 Use of Color.  Though I fear this suffers the same ambiguity and subjectivity as "discernible".

First, The reason I recommended "discernible" is because it would indicate the requirement that the focus indicator needs to be both "recognizable" and "perceivable".  This is much different than something that is "visible".  Something that is "visible" just needs to be in the range of wavelengths that the eye is sensitive to. A soldier wearing camouflage is "visible" with his surroundings, though he may not be "discernible" from his surroundings. In the same way, a white dot on a white background is "visible" with it's surroundings, but it is not "discernible" from it's surroundings. Or a repeating pattern that masks the change in state would be "visible", but not "discernible" due to it's camouflaged nature. "Visibility" does not require contrast or shape or difference from surroundings, but the ability to "discern" does. So I recommended at least changing that wording to be more in line with what we actually mean. I believe we want a "discernible" keyboard focus indicator, not just a "visible" one.

Second, it seems to me that there is a gap when it comes to the use of "pattern and imagery" to identify focus.  In this case, the pattern allows it to meet the requirement, but does not provide a distinctive indicator. The use of a camouflaging technique masks the change in state. Unfortunately, using the pattern shift and colors that are at least 3:1 contrast will also pass 1.4.1 - Use of Color, so other criterion don't help us close the gap here.  So we need to explicitly call out these nefarious ways to pass the criterion.
This may also help when the focus indicator is overlaid on something like an image/video or within an  image button that is not a solid color.

This _definitely_ needs some refining, but here's my first try at a definition for this... help would be appreciated.

Add a new requirement below "Contrast or thickness":
**Pattern or imagery:**  When patterns or imagery are used to distinguish the **focus indication area** of a User Interface Control must be discernible. At least one of the following is true:
- If a pattern is used, then it must not swap the position of colors with adjacent colors within the pattern on focus or hover.  [This needs work. It only covers swapping the position of current colors. It does not cover a gap where they could simply change colors of the checkers to different color values which still meet the 3:1 contrast ratio (passing 1.4.1), but would be equally as camouflaged]
- If an image is used... [I'm not sure what criterion to provide here, HELP!]

[Definition] Discernible: A means of presenting information, state, or distinguishing a visual element that is "perceivable" and "recognizable" from it's adjacent colors and patterns.

**Comment by JAWS-test** [comment]
I am unsure whether we really need a precise description of what is allowed and what is not allowed for each case of a possible focus display. For example, the focus display with pattern seems to me to be a good example of a gap, but at the same time very unlikely that something like this will ever be used as a focus indicator. The new SC for focus display is already complicated enough, so I wouldn't like to make it even more difficult. 

Perhaps it might be useful to define in the SC that a focus must be easily recognizable - and also to define minimum conditions. Then a tester can also detect a violation of the SC if the minimum conditions are met, but the focus is still poorly discernible (as with pattern focus).

Alternatively, the Understanding can provide information on good and bad focus indicators

**Comment by alastc** [comment]
For visible/discernible, as we can't change the WCAG 2.0 criteria I think it would be best to stay consistent between the two SCs.

For the pattern aspect, my concern is the increase in complexity relative to the likelihood of people actually doing this. (Tree falling in the woods sort of thing.)

It looks like it takes a lot of work to achieve (so, um, well done for working that out!), but have you ever seen this sort of approach 'in the wild'? I know I haven't, and I use focus styles a lot...

**Comment by alastc** [comment]
Proposed response:
<hr>
Whilst this is a gap in the definition, the complexity that would be introduced to plug that gap does not appear to be worth it. If the increased requirement brought in by the new success criteria leads people to try and get around it, we can revisit the issue in a future version of WCAG.

**Comment by alastc** [comment]
Hi @JasonAment,

Like the other issue, the working group [discussed this today](https://www.w3.org/2020/06/30-ag-minutes.html#item05) and agreed that:
<hr>
This is not a common pattern that we have seen in the wild, and it seems unlikely that the new success criteria would cause people to try this method. Whilst it is a potential hole, we would rather not complicate the text in order to plug that (quite rare) hole. We will [update the understanding document ](https://github.com/w3c/wcag/pull/1121/commits/197613a3abb9bf8c93e315ae27745e8eb2b5f07a#diff-79e164d9a49dfaa9222ec7823904bad6L157)to acknowledge this as a possible outcome.
<hr>
